group/assignment5/Assignment5.java

```diff
@@ -0,0 +1,149 @@
+package assignment5;

```

Comment:
I would add at the beginning a short description of the approach (i.e - will use topological sort and your defined graphs). Your solution is nice and clear, good work. Great job at doing the optional challenge!